.github/workflows/unit-tests.yaml

```diff
@@ -0,0 +1,200 @@
+name: Unit Tests
+
+# Execute workflow for each PR and with each merge to the trunk
+on:
+  pull_request:
+  push:
+    branches:
+      - main
+
+# Cancel the workflow if any new changes pushed to a feature branch or the trunk
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: false # for now, false (put jobs in queue) allowing multiple PR testing consequently
+env:
+  VYPER_DOWNLOAD_URL: "https://github.com/vyperlang/vyper/releases/download"
+  VYPER_VERSION: "v0.3.10"
+  VYPER_DOWNLOAD_FILENAME: "vyper.0.3.10+commit.91361694"
+  CARGO_NET_GIT_FETCH_WITH_CLI: true
+  UNIT_TESTS_RESULTS_XML: unit-tests-results.xml
+  RUSTC_BOOTSTRAP: 1 # Use rustc in bootstrap mode allowing cargo2junit to use experimental --json feature
+
+jobs:
+  unit-tests-macos:
+    strategy:
+      matrix:
+        include:
+          - name: "MacOS x86"
+            runner: macos-12-large
+          - name: "MacOS arm64"
+            runner: [self-hosted, macOS, ARM64]
+    runs-on: ${{ matrix.runner }}
+    name: ${{ matrix.name }}
+    defaults:
+      run:
+        shell: zsh {0} # For MacOS hosts, .zprofile is pre-configured
+    env:
+      VYPER_DOWNLOAD_EXTENSION: ".darwin"
+    steps:
+      # TODO: This step should be done as part of github runners hooks
+      # Users should always expect clean workspace when a new job starts
+      # https://github.blog/changelog/2022-04-04-github-actions-job-management-hooks-for-self-hosted-runners/
+      - name: Cleanup workspace
+        if: ${{ matrix.name }} == "MacOS ARM64"
+        run: |
+          setopt rmstarsilent
+          setopt +o nomatch
+          rm -rf ${{ github.workspace }}/*
+      - name: Checkout source
+        uses: actions/checkout@v4
+      - name: Prepare environment
+        run: |
+          brew install cmake ninja
+      - name: Build LLVM framework
+        run: |
+          cargo install compiler-llvm-builder
+          zkevm-llvm clone
+          zkevm-llvm build
+      - name: Install vyper compiler
+        run: |
+          curl --verbose \
+            --location -o "${PWD}/vyper" \
+            "${VYPER_DOWNLOAD_URL}/${VYPER_VERSION}/${VYPER_DOWNLOAD_FILENAME}${VYPER_DOWNLOAD_EXTENSION}" && \
+            chmod a+x "${PWD}/vyper"
+          echo "${PWD}" >> "${GITHUB_PATH}"
+      - name: Build zkvyper
+        run: |
+          cargo build -vv --release
+          echo "${PWD}/target/release" >> "${GITHUB_PATH}"
+      - name: Run unit tests
+        run: |
+          cargo install cargo2junit
```

Comment:
I’d suggest splitting everything but actually running the unit tests from the “Run unit tests” phase, to avoid cruft like that below in its section of the log.  I’d like to be able to click on that disclosure triangle in the logs and only see warnings I care about.

> Run cargo install cargo2junit
    Updating crates.io index
     Ignored package `cargo2junit v0.1.13` is already installed, use --force to override
   Compiling libc v0.2.152


https://github.com/matter-labs/era-compiler-vyper/actions/runs/7653852412/job/20856599305#step:8:1

Reply:
Fixed this one, thanks.